<?xml version="1.0" encoding="UTF-8"?>
    <source package="java.lang">
        <class name="ExceptionInInitializerError" line="28">
            <extends class="LinkageError"/>
            <javadoc line="28">
                Signals that an unexpected exception has occurred in a static initializer.
                  An &lt;code&gt;ExceptionInInitializerError&lt;/code&gt; is thrown to indicate that an
                  exception occurred during evaluation of a static initializer or the
                  initializer for a static variable.
                  &lt;p&gt;As of release 1.4, this exception has been retrofitted to conform to
                  the general purpose exception-chaining mechanism.  The &quot;saved throwable
                  object&quot; that may be provided at construction time and accessed via
                  the {@link #getException()} method is now known as the &lt;i&gt;cause&lt;/i&gt;,
                  and may be accessed via the {@link Throwable#getCause()} method, as well
                  as the aforementioned &quot;legacy method.&quot;                
                <author>
                    Frank Yellin                    
                </author>
                <since>
                    JDK1.1                    
                </since>
            </javadoc>
            <declaration name="serialVersionUID" type="long" line="45"/>
            <javadoc line="45">
                Use serialVersionUID from JDK 1.1.X for interoperability                
            </javadoc>
            <declaration name="exception" type="Throwable" line="50"/>
            <javadoc line="50">
                This field holds the exception if the
                  ExceptionInInitializerError(Throwable thrown) constructor was
                  used to instantiate the object                
                <serial>
                                        
                </serial>
            </javadoc>
            <javadoc line="60">
                Constructs an &lt;code&gt;ExceptionInInitializerError&lt;/code&gt; with
                  &lt;code&gt;null&lt;/code&gt; as its detail message string and with no saved
                  throwable object.
                  A detail message is a String that describes this particular exception.                
            </javadoc>
            <method name="ExceptionInInitializerError" type="constructor" line="66">
                <comment line="67">
                    Disallow subsequent initCause                    
                </comment>
            </method>
            <javadoc line="70">
                Constructs a new &lt;code&gt;ExceptionInInitializerError&lt;/code&gt; class by
                  saving a reference to the &lt;code&gt;Throwable&lt;/code&gt; object thrown for
                  later retrieval by the {@link #getException()} method. The detail
                  message string is set to &lt;code&gt;null&lt;/code&gt;.                
                <param>
                    thrown The exception thrown                    
                </param>
            </javadoc>
            <method name="ExceptionInInitializerError" type="constructor" line="78">
                <params>
                    <param name="thrown" type="Throwable"/>
                </params>
                <comment line="79">
                    Disallow subsequent initCause                    
                </comment>
            </method>
            <javadoc line="83">
                Constructs an ExceptionInInitializerError with the specified detail
                  message string.  A detail message is a String that describes this
                  particular exception. The detail message string is saved for later
                  retrieval by the {@link Throwable#getMessage()} method. There is no
                  saved throwable object.                
                <param>
                    s the detail message                    
                </param>
            </javadoc>
            <method name="ExceptionInInitializerError" type="constructor" line="93">
                <params>
                    <param name="s" type="String"/>
                </params>
                <comment line="95">
                    Disallow subsequent initCause                    
                </comment>
            </method>
            <javadoc line="98">
                Returns the exception that occurred during a static initialization that
                  caused this error to be created.
                  &lt;p&gt;This method predates the general-purpose exception chaining facility.
                  The {@link Throwable#getCause()} method is now the preferred means of
                  obtaining this information.                
                <return>
                    the saved throwable object of this
                      <code>ExceptionInInitializerError</code>, or <code>null</code>
                      if this <code>ExceptionInInitializerError</code> has no saved
                      throwable object.                    
                </return>
            </javadoc>
            <method name="getException" type="Throwable" line="111"/>
            <javadoc line="115">
                Returns the cause of this error (the exception that occurred
                  during a static initialization that caused this error to be created).                
                <return>
                    the cause of this error or <code>null</code> if the
                      cause is nonexistent or unknown.                    
                </return>
                <since>
                    1.4                    
                </since>
            </javadoc>
            <method name="getCause" type="Throwable" line="123"/>
        </class>
    </source>